thresholdkey.go

```diff
@@ -8,27 +8,47 @@ import (
 	"math/big"
 )
 
+// Public key for a threshold Paillier scheme.
+//
+// `V` is a generator in  the cyclic group of squares Z_n^2 and is used to
+// execute a zero-knowledge proof of a received share decryption.
+//
+// `Vi` is an array of verification keys for each decryption server `i` used to
+// execute a zero-knowledge proof of a received share decryption.
+//
+// Key generation, encryption, share decryption and combining for the threshold
+// Paillier scheme has been described in [DJN 10], section 5.1.
+//
+//     [DJN 10]: Ivan Damgard, Mads Jurik, Jesper Buus Nielsen, (2010)
+//               A Generalization of Paillier’s Public-Key System
+//               with Applications to Electronic Voting
+//               Aarhus University, Dept. of Computer Science, BRICS
 type ThresholdKey struct {
 	PublicKey
 	TotalNumberOfDecryptionServers int
 	Threshold                      int
-	V                              *big.Int
-	Vi                             []*big.Int
-	G                              *big.Int // usually G is set to N+1
+	V                              *big.Int   // needed for ZKP
+	Vi                             []*big.Int // needed for ZKP
 }
 
-// returns the value of (4*delta**2)** -1  mod n
-func (this *ThresholdKey) CombineSharesConstant() *big.Int {
-	tmp := new(big.Int).Mul(FOUR, new(big.Int).Mul(this.Delta(), this.Delta()))
+// Returns the value of [(4*delta^2)]^-1  mod n.
+// It is a constant value for the given `ThresholdKey` and is used in the last
+// step of share combining.
+func (this *ThresholdKey) combineSharesConstant() *big.Int {
+	tmp := new(big.Int).Mul(FOUR, new(big.Int).Mul(this.delta(), this.delta()))
 	return (&big.Int{}).ModInverse(tmp, this.N)
 }
 
-// returns the factorial of the number of TotalNumberOfDecryptionServers
-func (this *ThresholdKey) Delta() *big.Int {
+// Returns the factorial of the number of `TotalNumberOfDecryptionServers`.
+// It is a contant value for the given `ThresholdKey`.
+func (this *ThresholdKey) delta() *big.Int {
 	return Factorial(this.TotalNumberOfDecryptionServers)
 }
 
-func (this *ThresholdKey) MakeVerificationBeforeCombiningPartialDecryptions(shares []*PartialDecryption) error {
+// Checks if the number of received, unique shares is less than the
```

Comment:
Though this matters less for unexported functions, Go prefers the form “<function name> checks if…”, i.e. `makeVerificationBeforeCombiningPartialDecryptions checks if…`.

Though this function itself seems a bit odd, if I'm being honest <_<

Reply:
You know what, I'll just rename it to `verifyPartialDecryptions` 😉

Reply:
So we should always start documentation for exported function with its name?

Reply:
Correct.